doc/manual/Checker_Library.md

```diff
@@ -0,0 +1,84 @@
+specific Use Case or in one specific environment. The checks may be distributed
+in multiple Checker Bundles in the Checker Library.
+
+Each Checker Library needs to have a documentation what is included. Here we
+have to distinguish between "Checks" and "Rules". A standard specify rules,
+which are described in textual form. A check is a test implementation which
+inspects the input document to verify if it is compliant to the standard.
+
+## Rules
+
+### Rule Characteristics
+
+- Rules are atomic, easy understandable and independent of how many rules are
+  necessary in the standard.
+- A Rule should be part of a standard, as it is today in many ASAM standards.
+- Rules can come from ASAM, but also from other organizations or can be company
+  internal rules.
+- A Checker Library may also need to document rules, if they are not explicitly
+  written in the according standard. This documentation can be used as a proposal
+  to the standardization project to improve the standard itself.
+
+### Rule Documentation Template
+
+If a Checker Library needs to define and document own rules, then the following
+template can be used.
+
+**Rule ID:** _The Rule ID should not change, if there is no change in the rule._
+
+**Description:** _Clear textual description of the rule._
+
+**Severity:** _mandatory or recommendation_
+
+**Version range:** _Information for which versions of the standard this rule applies._
+
+## Checks
+
+### Check Characteristics
+
+- Ideally, a check addresses exactly one rule. In reality, a check can address
+  several rules, or several checks might be necessary to check a certain rule.
+- For each check, at least one example input file needs to be provided, which
+  shows the rule violation. Additionally a file including the fix of the
+  violation shall be provided.
+
+### Check Documentation Template
+
+**Check ID:** _The ID shall be unique within the Checker Library and
+self-explanatory as possible._
+
+**Version:** _The check itself has also a version number for identifying
+changes in the results, which are based on implementation changes._
+
+**Description:** _Clear textual description of the check._
+
+**Parameters:** _Description of the parameters of this check. For a
+well-defined standard, there should be no parameter necessary. It's often the
+case that some are needed._
+
+**Addressed Rule:** _Link to the standard or to the rule definition of the
+Checker Library, if not appropriate rule is explicitly written in the
+standard._
+
+**Version range:** _Information how this check is implemented for which
+versions._
+
+**Approval Information:** _Who has approved and when was this check approved.
+This may be empty if the check is not approved yet._
+
+**Metadata:** _A check may log arbitrary meta data (key-value pairs) for
+traceability. Example: If a check is implemented as a behavior evaluation of a
+simulation, then the check needs to log the version and configuration of the
+simulator._
```

Comment:
```suggestion
**Metadata:** _A check may log arbitrary meta data during execution (e.g., for
traceability). The meta data produced by a check shall be included and described 
in its Check Documentation, (e.g., if a check logs in metadata version and configuration of 
a simulator, the checker documentation shall describe the content of aforementioned metadata fields)._
```

In this section of the manual, I think it is crucial to emphasize that metadata shall be well documented.